metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/RdfDeserializerImpl.java

```diff
@@ -321,27 +316,69 @@ Map<String, ResourceInfo> getResourceEntries(Document document,
 
     // For each resource, check whether they are configured for oEmbed.
     final Map<String, ResourceInfo> result = HashMap.newHashMap(urls.size());
-    final Set<String> oEmbedUrls = getOEmbedUrls(document);
-    final Set<String> iiifUrls = getIIFUrls(document);
+    final Map<String, CachedHasServiceValue> oEmbedUrls = getResourceUrls(document, getOEmbedExpression);
+    final Map<String, CachedHasServiceValue> iiifUrls = getResourceUrls(document, getIIIFExpression);
     for (Entry<String, Set<UrlType>> entry : urls.entrySet()) {
-
-      RdfResourceKind rdfResourceKind;
-      if (oEmbedUrls.contains(entry.getKey())) {
+      final RdfResourceKind rdfResourceKind;
+      final String hasServiceValue;
+      if (oEmbedUrls.containsKey(entry.getKey())) {
         rdfResourceKind = RdfResourceKind.OEMBEDDED;
+        hasServiceValue = oEmbedUrls.get(entry.getKey()).getValue();
+      } else if (iiifUrls.containsKey(entry.getKey())) {
+        rdfResourceKind = RdfResourceKind.IIIF;
+        hasServiceValue = iiifUrls.get(entry.getKey()).getValue();
       } else {
-        if (iiifUrls.contains(entry.getKey())) {
-          rdfResourceKind = RdfResourceKind.IIIF;
-        } else {
-          rdfResourceKind = RdfResourceKind.STANDARD;
-        }
+        rdfResourceKind = RdfResourceKind.STANDARD;
+        hasServiceValue = null;
       }
-      result.put(entry.getKey(), new ResourceInfo(entry.getValue(), rdfResourceKind));
+      result.put(entry.getKey(), new ResourceInfo(entry.getValue(), rdfResourceKind, hasServiceValue));
     }
 
     // Done
     return result;
   }
 
-  record ResourceInfo(Set<UrlType> urlTypes, RdfResourceKind rdfResourceKind) {
+  static class CachedHasServiceValue {
```

Comment:
I am wondering if this is actually required. During the getResourceUrls does the check already go through the trouble of checking if there is a WebResource with a matching rdfAbout? So perhaps we could just check in that match if there is a service and avoid a second lookup?